### PR TITLE
add BE udf bitmap xor

### LINKED ISSUE #2536: support bitmap xor
sytax
```
BITMAP bitmap_xor(BITMAP map1, BITMAP map2)
```

this function will return the xor result of inputs two bitmap .

#2533

### PR DESCRIPTION
**Pull request by lihuigang**
## Proposed changes

this function will return the xor result of inputs two bitmap .

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #2536), and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — COMMENTED** [review]
Hi, could you please add document for this new function?

**Comment by lihuigang** [comment]
Yes ,I'd like to

**Comment by lihuigang** [comment]
hi,I've added documents

File: be/src/exprs/bitmap_function.cpp [hunk 1]
```diff
@@ -477,6 +477,27 @@ StringVal BitmapFunctions::bitmap_and(FunctionContext* ctx, const StringVal& lhs
     return serialize(ctx, &bitmap);
 }
 
+StringVal BitmapFunctions::bitmap_xor(FunctionContext* ctx, const StringVal&  lhs,
```
Comment (morningman) [inline be/src/exprs/bitmap_function.cpp:480]:
```suggestion
StringVal BitmapFunctions::bitmap_xor(FunctionContext* ctx, const StringVal& lhs,
```

File: .rat-excludes [hunk 2]
```diff
@@ -1,33 +0,0 @@
-.*json
```
Comment (morningman) [inline .rat-excludes:1]: Why delete this file?

File: be/src/util/bitmap_value.h [hunk 3]
```diff
@@ -1087,6 +1088,63 @@ class BitmapValue {
...
+            case SINGLE:
+                switch (_type) {
+                    case EMPTY:
+                        add(rhs._sv);
+                        break;
+                    case SINGLE:
+                        if (_sv == rhs._sv) {
+                            _type = EMPTY;
+                            _bitmap.clear();
+                        }else{
```
Comment (morningman) [inline be/src/util/bitmap_value.h:1109]:
```suggestion
                        } else {
```

File: .rat-excludes (hunk 2 above)
Comment (lihuigang) [reply .rat-excludes]: Sorry, I accidentally deleted

Comment (lihuigang) [reply .rat-excludes]: 您好，这个文件已经恢复了~

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**